src/7.1_HW/Company.ts

```diff
@@ -0,0 +1,17 @@
+import {Department} from "./Department";
+import {PreHiredEmployee} from "./PreHiredEmployee";
+import {Employee} from "./Employee";
+
+export class Company {
+    name: string;
+    departments: Department[] = [];
+    preHiredStaff: PreHiredEmployee[];
+    allStaff: Employee[];
```

Comment:
This property should contain a data structure with both Employee and PreHiredEmployee entities. Then you won't need the `preHiredStaff` property.